### PR TITLE
feat: add goreleaser to publish binaries

### PR DESCRIPTION
**Pull request by eytan-avisror**
Signed-off-by: Eytan Avisror <eytan_avisror@intuit.com>

Adds to #30

### CODE REVIEW

**Review by garomonegro — APPROVED** [review]

**Review by vgunapati — APPROVED** [review]

**Review by ccfishk — APPROVED** [review]
lgtm

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      - name: Checkout
+        uses: actions/checkout@master
+
+      - name: Setup Go
+        uses: actions/setup-go@v1
+        with:
+          go-version: ^1.17
+
+      - name: GoReleaser
+        uses: goreleaser/goreleaser-action@v1
```
Comment (kevdowney) [inline .github/workflows/release.yaml:21]: This is at `goreleaser/goreleaser-action@v2` now.

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+jobs:
+  releases-matrix:
+    name: Release Go Binary
+    runs-on: ubuntu-18.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@master
+
+      - name: Setup Go
+        uses: actions/setup-go@v1
```
Comment (kevdowney) [inline .github/workflows/release.yaml:16]: This is also at `v2`

Comment (kevdowney) [reply .github/workflows/release.yaml:16]: Fyi. You can setup `dependabot` to auto update the actions.

https://github.com/keikoproj/addon-manager/blob/master/.github/dependabot.yml

File: .goreleaser.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    main: ./
+    binary: aws-auth
+    env:
+      - CGO_ENABLED=0
+    goos:
+      - linux
+      - windows
+      - darwin
+    ldflags:
+      - -s -w -X github.com/keikoproj/aws-auth/cmd/cli.gitCommit={{.Commit}} -X github.com/keikoproj/aws-auth/cmd/cli.buildDate={{.Date}} -X github.com/keikoproj/aws-auth/cmd/cli.pkgVersion={{.Version}}
```
Comment (kevdowney) [inline .goreleaser.yaml:18]: Addon manager uses goreleaser too https://github.com/keikoproj/addon-manager/blob/master/.goreleaser.yml

The `ldflags` can be multi line

**Review by kevdowney — APPROVED** [review]
LGTM, minor comments.

**Pull request merged**